artemis-journal/pom.xml

```diff
@@ -15,57 +15,81 @@
+    </parent>
 
-   <artifactId>artemis-journal</artifactId>
-   <packaging>jar</packaging>
-   <name>ActiveMQ Artemis Journal</name>
+    <artifactId>artemis-journal</artifactId>
+    <packaging>jar</packaging>
+    <name>ActiveMQ Artemis Journal</name>
 
-   <properties>
-      <activemq.basedir>${project.basedir}/..</activemq.basedir>
-   </properties>
+    <properties>
+        <activemq.basedir>${project.basedir}/..</activemq.basedir>
+    </properties>
 
-   <dependencies>
-      <dependency>
-         <groupId>org.jboss.logging</groupId>
-         <artifactId>jboss-logging-processor</artifactId>
-         <scope>provided</scope>
-         <optional>true</optional>
-      </dependency>
+    <dependencies>
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging-processor</artifactId>
+            <scope>provided</scope>
+            <optional>true</optional>
+        </dependency>
 
-      <!--
-          JBoss Logging
-      -->
-      <dependency>
-         <groupId>org.jboss.logging</groupId>
-         <artifactId>jboss-logging</artifactId>
-      </dependency>
-      <dependency>
-         <groupId>org.jboss.logmanager</groupId>
-         <artifactId>jboss-logmanager</artifactId>
-         <scope>test</scope>
-      </dependency>
-      <dependency>
-         <groupId>org.apache.activemq</groupId>
-         <artifactId>artemis-commons</artifactId>
-         <version>${project.version}</version>
-      </dependency>
-      <dependency>
-         <groupId>org.apache.activemq</groupId>
-         <artifactId>artemis-native</artifactId>
-         <version>${project.version}</version>
-      </dependency>
-      <!-- needed to compile the tests -->
-      <dependency>
-         <groupId>junit</groupId>
-         <artifactId>junit</artifactId>
-         <scope>test</scope>
-      </dependency>
-   </dependencies>
-</project>
+        <!--
+            JBoss Logging
+        -->
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.logmanager</groupId>
+            <artifactId>jboss-logmanager</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-commons</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-native</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <!-- needed to compile the tests -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>net.openhft</groupId>
```

Comment:
Hi @franz1981   

It looks like the tests you've added are unit tests contained within the artemis-journal module.  For this I would add the test deps to the top level pom.xml.  The tests/pom.xml is used for a bunch of integration tests.  I think going forward we should keep everything in the top level pom.xml.  I'll update the hacking guide, my apologies that this wasn't clear.

Here is what I would do:
1. Add JMH dep under the dependencyManagement section of the main pom.xml.
2. Exclude the new tests in the default profile.
3. Create a new profile "extraTests" in the artemis-journal and use this to enable the additional tests and dependencies you've added.

This process is less than ideal, but after some discussion with Apache Legal and other community members we decided this was the best approach to ensure we are always compliant with Apache Licensing.

I'm around on IRC if you want to chat "#activemq"

Thank you